index.js

```diff
@@ -23,15 +25,42 @@ function createFunction(parameters, defaultType) {
             throw new Error('Unknown function type "' + type + '"');
         }
 
-        if (property === '$zoom') {
+        if (zoomAndFeatureDependent) {
+            var featureFunctions = {};
+            var featureFunctionStops = [];
+            for (var s = 0; s < parameters.stops.length; s++) {
+                var stop = parameters.stops[s];
+                if (featureFunctions[stop[0].zoom] === undefined) {
+                    featureFunctions[stop[0].zoom] = {
+                        type: parameters.type,
+                        property: parameters.property,
+                        stops: []
+                    };
+                }
+                featureFunctions[stop[0].zoom].stops.push([stop[0].value, stop[1]]);
+            }
+
+            for (var z in featureFunctions) {
+                featureFunctionStops.push([parseInt(z), createFunction(featureFunctions[z])]);
+            }
+            fun = function(zoom, feature) {
+                return evaluateCategoricalFunction({ stops: featureFunctionStops, base: parameters.base }, zoom)(zoom, feature);
```

Comment:
`evaluateCategoricalFunction` doesn't accept a `base` parameter. 

Why is `evaluateCategoricalFunction` the right function to use here? How do you interpolate across zooms?